ngraph/core/src/op/experimental_detectron_prior_grid_generator.cpp

```diff
@@ -0,0 +1,154 @@
+    visitor.on_attribute("w", m_attrs.w);
+    visitor.on_attribute("stride_x", m_attrs.stride_x);
+    visitor.on_attribute("stride_y", m_attrs.stride_y);
+    return true;
+}
+
+shared_ptr<Node> op::v6::ExperimentalDetectronPriorGridGenerator::clone_with_new_inputs(
+    const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+
+    return make_shared<op::v6::ExperimentalDetectronPriorGridGenerator>(
+        new_args.at(0), new_args.at(1), new_args.at(2), m_attrs);
+}
+
+static constexpr size_t priors_port = 0;
+static constexpr size_t featmap_port = 1;
+static constexpr size_t im_data_port = 2;
+
+void op::v6::ExperimentalDetectronPriorGridGenerator::validate()
+{
+    auto priors_shape = get_input_partial_shape(priors_port);
+    auto featmap_shape = get_input_partial_shape(featmap_port);
+    auto im_data_shape = get_input_partial_shape(im_data_port);
+
+    if (priors_shape.rank().is_dynamic() || featmap_shape.rank().is_dynamic())
+    {
+        return;
+    }
+
+    NODE_VALIDATION_CHECK(
+        this, priors_shape.rank().get_length() == 2, "Priors rank must be equal to 2.");
+
+    NODE_VALIDATION_CHECK(this,
+                          priors_shape[1].is_static() && priors_shape[1].get_length() == 4u,
+                          "The last dimension of the 'priors' input must be equal to 4. Got: ",
+                          priors_shape[1]);
+
+    NODE_VALIDATION_CHECK(
+        this, featmap_shape.rank().get_length() == 4, "Feature_map rank must be equal to 4.");
+
+    if (im_data_shape.rank().is_dynamic())
+    {
+        return;
+    }
+
+    NODE_VALIDATION_CHECK(
+        this, im_data_shape.rank().get_length() == 4, "Im_data rank must be equal to 4.");
+
+    const auto num_batches_featmap = featmap_shape[0];
+    const auto num_batches_im_data = im_data_shape[0];
+    NODE_VALIDATION_CHECK(this,
+                          num_batches_featmap.same_scheme(num_batches_im_data),
+                          "The first dimension of both 'feature_map' and 'im_data' must match. "
+                          "Feature_map: ",
+                          num_batches_featmap,
+                          "; Im_data: ",
+                          num_batches_im_data);
+}
+
+void op::v6::ExperimentalDetectronPriorGridGenerator::validate_and_infer_types()
+{
+    auto priors_shape = get_input_partial_shape(priors_port);
+    auto featmap_shape = get_input_partial_shape(featmap_port);
+    auto input_et = get_input_element_type(0);
+
+    validate();
+
+    set_output_size(1);
+    PartialShape out_shape = {Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic(), 4};
+    if (m_attrs.flatten)
+    {
+        out_shape = PartialShape{Dimension::dynamic(), 4};
+    }
+
+    if (priors_shape.rank().is_dynamic() || featmap_shape.rank().is_dynamic())
+    {
+        set_output_type(0, input_et, out_shape);
+        return;
+    }
+
+    if (priors_shape[0].is_dynamic() || featmap_shape[2].is_dynamic() ||
```

Comment:
Could you add variables for priors_shape[0], featmap_shape[2] and featmap_shape[3] - they used in several places below?

Reply:
Done.